Update package.json

ed-wright opened the pull request:
Update peer dependencies to solve issue #22 and allow integration with FontAwesome 6

ed-wright committed:
Update package.json

jimappleg8 commented:
The change that was made to the package.json file did not seem to resolve the issue. The later part of the range was removed (i.e. "^1.2.0 || ^6.1.0" changed to "^1.2.0") which I think still excludes anything past version 1.2.x of the fontawesome-svg-core file. 

I'm not an expert, but I think what needs to change is either to allow higher versions of 6 (i.e. 6.2.0 is the latest) 

"@fortawesome/fontawesome-svg-core": "^1.2.0 || ^6.0.0"

or just allow anything greater than 1.2.0:

"@fortawesome/fontawesome-svg-core": ">1.2.0"

I'm getting this info from here: https://github.com/npm/node-semver#versions

ed-wright commented:
@jimappleg8 not an expert either :) im checking this now as I believe that you are mostly correct but the "@fortawesome/fontawesome-svg-core": ">1.2.0" part I am unsure about but this may be my misunderstanding.

ed-wright commented:
@jimappleg8 i just re-checked out the code and installed via Yarn. I have also updated the package.json and lockfile using npm.

I also used [https://semver.npmjs.com/](https://semver.npmjs.com/) to check and the string i have seems okay although i have updated it to be more permissive. (^6.0.0 instead of ^6.1.0)

![image](https://user-images.githubusercontent.com/5090163/190703789-d0a49bd4-b2cb-4ae3-abc2-2af3bfa937bc.png)

If you install via yarn you will currently get...

![image](https://user-images.githubusercontent.com/5090163/190704380-24938518-0c89-4d02-9be4-b84a3fe4d85f.png)

if you install via NPM you get.,..

![image](https://user-images.githubusercontent.com/5090163/190704481-bf735713-82fc-41af-9170-8d684da9517a.png)

This is not good, looking into the fix...

ed-wright committed:
updates

ed-wright committed:
Update package.json

ed-wright commented:
@jimappleg8 @lapc002 I have had a PR open for a long time now attempting to fix this. I have now published a fork of this repo on NPM

https://github.com/ed-wright/gatsby-plugin-fontawesome-css
https://www.npmjs.com/package/gatsby-plugin-fontawesome-css-2

jzabala commented:
Hey @ed-wright I am very sorry. I don't have time anymore to keep maintaining this project. Let me know if you would like me to transfer it to you.

Thanks :)

ed-wright commented:
Yes please, could you also transfer the npm package ownership too

jzabala commented:
Hi @ed-wright thank you very much for offering to take control of the project 🙏🏼. I see this error when I try to transfer the github project to you:

<img width="718" alt="Screen Shot 2022-10-14 at 8 44 59 PM" src="https://user-images.githubusercontent.com/1315054/195961615-c784ecdb-4058-4d2e-a473-bde2df7c53c6.png">

I already sent to you the invitation to be a maintainer in `npm`:

<img width="794" alt="Screen Shot 2022-10-14 at 8 49 40 PM" src="https://user-images.githubusercontent.com/1315054/195961514-cea488d8-a955-4f65-8b1e-afe38041aa69.png">

ed-wright commented:
@jzabala sorted sorry about that :)

jzabala commented:
hey @ed-wright no idea what is going on hahah. Do you understand this error?
<img width="711" alt="Screenshot 2022-11-08 at 2 23 28 PM" src="https://user-images.githubusercontent.com/1315054/200645362-a7ec29d0-0bad-4124-b5b2-6f8441e983e5.png">

In the meantime I added you as a collaborator of the repo.
